tests/test-team/template-mgmt-component-tests/template-mgmt-create-nhs-app-template-page.component.ts

```diff
@@ -32,15 +50,193 @@ test.describe('Create NHS App Template Page', () => {
+    ).toHaveText(['Enter a template message']);
+  });
+
+  test('Validate error messages on the create NHS App message template page with no template name', async ({
+    page,
+    baseURL,
+  }) => {
+    const createTemplatePage = new TemplateMgmtCreatePage(page);
+
+    await createTemplatePage.loadPage(emptySessionData.id);
 
     await expect(page).toHaveURL(
-      `${baseURL}/templates/create-nhs-app-template/${nhsAppNoTemplateSessionData.id}`
+      `${baseURL}/templates/create-nhs-app-template/${emptySessionData.id}`
     );
     expect(await createTemplatePage.pageHeader.textContent()).toBe(
       'Create NHS App message template'
     );
+    const templateMessage = 'Test Message box';
+    await page.locator('[id="nhsAppTemplateMessage"]').fill(templateMessage);
+    await createTemplatePage.clickContinueButton();
+    await expect(page.locator('.nhsuk-error-summary')).toBeVisible();
+
+    await expect(
+      page.locator('ul[class="nhsuk-list nhsuk-error-summary__list"] > li')
+    ).toHaveText(['Enter a template name']);
+  });
+
+  test('5000 words Entered in Template body and word count correctly displayed', async ({
+    page,
+    baseURL,
+  }) => {
+    const createTemplatePage = new TemplateMgmtCreatePage(page);
+
+    await createTemplatePage.loadPage(emptySessionForTemplateCreation.id);
+
+    await expect(page).toHaveURL(
+      `${baseURL}/templates/create-nhs-app-template/${emptySessionForTemplateCreation.id}`
+    );
+    const templateName = 'NHS Testing 123';
+    await page.locator('[id="nhsAppTemplateName"]').fill(templateName);
+    const templateMessage = 'T'.repeat(5000).trim();
+    await page.locator('[id="nhsAppTemplateMessage"]').fill(templateMessage);
+    await expect(page.getByText('5000 of 5000 characters')).toBeVisible();
+  });
+
+  test('5001 words attempted to be entered in Template body and only 5000 allowed', async ({
+    page,
+    baseURL,
+  }) => {
+    const createTemplatePage = new TemplateMgmtCreatePage(page);
+
+    await createTemplatePage.loadPage(emptySessionForTemplateCreation.id);
+
+    await expect(page).toHaveURL(
+      `${baseURL}/templates/create-nhs-app-template/${emptySessionForTemplateCreation.id}`
+    );
+    const templateName = 'NHS Testing 123';
+    await page.locator('[id="nhsAppTemplateName"]').fill(templateName);
+    const templateMessage = 'T'.repeat(5001).trim();
+    await page.locator('[id="nhsAppTemplateMessage"]').fill(templateMessage);
+    await expect(page.getByText('5000 of 5000 characters')).toBeVisible();
+  });
+
+  const detailsSections = [
+    '[data-testid="personalisation-details"]',
+    '[data-testid="lines-breaks-and-paragraphs-details"]',
+    '[data-testid="headings-details"]',
+    '[data-testid="bold-text-details"]',
+    '[data-testid="link-and-url-details"]',
+    '[data-testid="naming-your-templates"]',
+  ];
+
+  for (const section of detailsSections) {
+    test(`personalisation mark expanding fields for ${section}`, async ({
```

Comment:
Could expand if you wanted to get into more detail by verifying visible text within those sections - this might be a ticket for later if the wording is subject to change but you could store the content of each component in your page object and refer to that to make sure the correct text is visible

You could also verify that you can also close the expanding fields as well

Reply:
I think the wording within the sections is static and covered by unit tests (as with the other static content), maybe something to verify there is text there? Agree that closing them is a good addition.